trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/Experiment.java

```diff
@@ -82,8 +84,14 @@ public Experiment(@JsonProperty("name") String name,
      * @return the experiment model
      */
     public static Experiment from(TmfExperiment experiment, UUID expUUID) {
-        List<UUID> traceUUIDs = ExperimentManagerService.getTraceUUIDs(expUUID);
-        Set<Trace> traces = new LinkedHashSet<>(Lists.transform(traceUUIDs, uuid -> Trace.from(TraceManagerService.getOrCreateTraceInstance(uuid), uuid)));
+        Iterator<UUID> traceUUIDs = ExperimentManagerService.getTraceUUIDs(expUUID).iterator();
+        Iterator<ITmfTrace> traceInstances = experiment.getTraces().iterator();
```

Comment:
The getTraces() returns an array of ITmfTrace. Each ITmfTrace might be another other TmfExperiment or trace with children. To get the actual traces you need to use TmfTraceManager.getTraceSet(experiment)) to get the leaf traces. However, TmfTraceManager.getTraceSet(experiment)) doesn't guarantee the same order. I fixed bug #51 where a UUID mismatch happened. This caused a problem in one of our non-open source downstream projects. 

Besides, relying that 2 collections return the same order is not ideal, because it can easily break by unrelated code changes. So, I think we'll need to be able to have a lookup from trace uuid to ITmfTrace similar to the first version of this PR.

Reply:
Good point. I agree. We can feed the TmfExperiment with traces in a certain order but we can't prevent it (or an subclass of TmfExperiment) from shuffling them around when returning getTraces().
Sorry for that extra back and forth, please revert to first version of PR while fixing the issues mentioned in earlier comments.

Reply:
Np, done.